tests/test_graph.py

```diff
@@ -0,0 +1,20 @@
+import pytest
+import torch
+import murenn
+
+
+@pytest.mark.parametrize("J", list(range(1, 3)))
+@pytest.mark.parametrize("Q", [3, 4])
+@pytest.mark.parametrize("T", [8, 16])
+@pytest.mark.parametrize("padding_mode", ["symmetric", "zeros"])
+def test_shape(J, Q, T, padding_mode):
+    B, C, N = 2, 3, 2**(J+4)
+    x = torch.zeros(B, C, N)
+    graph = murenn.MURENN_GRAPH(
+        J=J,
+        Q=Q,
+        T=T,
+        padding_mode=padding_mode,
+    )
+    y = graph(x)
+    assert y.shape == (B, C*Q, J, 2**4)
```

Comment:
update shape

Reply:
test differentiability: require grad in the weights of the Conv1D's
define a loss function based on the output of MuReNNDirect. For example: the sum of all coefficients
Compute `w.grad`
It should be float32, not `None`